cpp/CMakeLists.txt

```diff
@@ -185,28 +185,14 @@ endif("$ENV{CONDA_BUILD}" STREQUAL "1")
 ###################################################################################################
 # - find arrow ------------------------------------------------------------------------------------
 
-find_path(ARROW_INCLUDE "arrow"
-          HINTS "$ENV{ARROW_ROOT}/include")
+message(STATUS "BUILDING ARROW")
+include(ConfigureArrow)
 
-find_library(ARROW_LIBRARY "arrow"
-             NAMES libarrow
-             HINTS "$ENV{ARROW_ROOT}/lib" "$ENV{ARROW_ROOT}/build")
-
-find_library(ARROW_CUDA_LIBRARY "arrow_cuda"
-             NAMES libarrow_cuda
-             HINTS "$ENV{ARROW_ROOT}/lib" "$ENV{ARROW_ROOT}/build")
-
-message(STATUS "ARROW: ARROW_INCLUDE set to ${ARROW_INCLUDE}")
-message(STATUS "ARROW: ARROW_LIBRARY set to ${ARROW_LIBRARY}")
-message(STATUS "ARROW: ARROW_CUDA_LIBRARY set to ${ARROW_CUDA_LIBRARY}")
-
-add_library(arrow SHARED IMPORTED ${ARROW_LIBRARY})
-add_library(arrow_cuda SHARED IMPORTED ${ARROW_CUDA_LIBRARY})
-
-if(ARROW_INCLUDE AND ARROW_LIBRARY AND ARROW_CUDA_LIBRARY)
-    set_target_properties(arrow PROPERTIES IMPORTED_LOCATION ${ARROW_LIBRARY})
-    set_target_properties(arrow_cuda PROPERTIES IMPORTED_LOCATION ${ARROW_CUDA_LIBRARY})
-endif(ARROW_INCLUDE AND ARROW_LIBRARY AND ARROW_CUDA_LIBRARY)
+if(ARROW_FOUND)
+    message(STATUS "Apache Arrow found in ${ARROW_INCLUDE_DIR}")
+else()
+    message(FATAL_ERROR "Apache Arrow not found, please check your settings.")
+endif(ARROW_FOUND)
```

Comment:
We shouldn't just roll this back entirely. We should put this behind a CMake variable of `ARROW_STATIC` or something along those lines to give users the ability to either statically link or dynamically link.

Reply:
Do we want to use the shared libraries from conda, or build them ourselves?

Reply:
A user could do either. Our current behavior is to just look in the CMake search path along with the `ARROW_ROOT` hint.

We shouldn't try to support building / distributing an arrow shared library alongside the libcudf shared library.

Reply:
Done.